### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by henrygab**
Remove two compiler warnings.

### CODE REVIEW

File: Adafruit_USBD_Device.cpp
Comment (hathach) [inline Adafruit_USBD_Device.cpp:324]: this is one actually a bug, not simply a warning.

File: Adafruit_USBD_Device.cpp
Comment (hathach) [inline Adafruit_USBD_Device.cpp]: @henrygab you forgot the bracket, it causes a bigger issue that -1 will always be returned. I think we actually can leave it as it is and have the caller init it rather. Though I am more than OK if you can put a bracket here as well.

```C
static int strcpy_utf16(const char *s, uint16_t *buf, int bufsize)
{
  int i = 0;
  int buflen = 0;

  while (s[i] != 0) {
    uint32_t codepoint = 0; // init here is better
````

**Review by hathach — COMMENTED** [review]
Thank you for PR as usual. There is a bit of issue with if/else without bracket. Once it is fixed, we could merge this.

**Comment by henrygab** [comment]
Hi @hathach,

I fixed the bracket, and also fixed other issues in this code.  For example, UTF-8 does not allow encoding of code points over 0x0010_FFFF, and there are a number of octets that are always illegal in UTF-8 encoding.  I have provided links to authority for each new illegal stream.

**_NOTE:  I do not have an easy way to test this code.  Thus, please do not merge without at least basic validation._**

Please feel free to send any questions.

**Comment by hathach** [comment]
> Hi @hathach,
> 
> I fixed the bracket, and also fixed other issues in this code. For example, UTF-8 does not allow encoding of code points over 0x0010_FFFF, and there are a number of octets that are always illegal in UTF-8 encoding. I have provided links to authority for each new illegal stream.
> 
> **_NOTE: I do not have an easy way to test this code. Thus, please do not merge without at least basic validation._**
> 
> Please feel free to send any questions.

I have no idea how to test these Unicode string. @kaysievers would you mind taking a look at the changes. Else I am afraid, we will leave this PR pending for quite some time.

**Comment by henrygab** [comment]
There is a useful table at https://github.com/cyb70289/utf8/#utf-8-coding-format.    UTF-8 decoding bugs have caused many serious issues.

I have not found a proper set of test data to validate a UTF-8 - to - codepoint decoder.

**Comment by henrygab** [comment]
@hathach  -- OK, I've now tested the UTF-8 decoder against **_all_** possible inputs.  I couldn't find a good test, so had to write it from scratch.  See [Gist 92a7077e](https://gist.github.com/henrygab/92a7077e4cc9b3e6b098ecaa5fa2d908).

The test tracks very little state.  It does this by relying on the fact that sorted codepoints will encode to sorted byte sequences ... this is why it can require a successful decoding to equal the prior decoded codepoint + 1.

Anyways, this should now be ready for your acceptance, as test code has now validated the correct decoding of all valid UTF-8 codepoints, and that no invalid UTF-8 will decode into a codepoint.  😁

**Comment by hathach** [comment]
@henrygab sorry, I was on vacation, will check this out on Monday.

File: Adafruit_USBD_Device.cpp
Comment (hathach) [inline Adafruit_USBD_Device.cpp:219]: too many texts :sweat_smile: :sweat_smile:

**Review by hathach — APPROVED** [review]
thank you very much for your great effort correcting and testing as always. Really great works so far, even writing tests yourself :+1: :+1:

**Pull request merged**
